store/displayPreferences.ts

```diff
@@ -1,116 +1,108 @@
-import Vue from 'vue';
 import { DisplayPreferencesDto } from '@jellyfin/client-axios';
-import { MutationTree, ActionTree, GetterTree } from 'vuex';
+import { MutationTree, ActionTree } from 'vuex';
+import { merge, toString as toStr } from 'lodash';
+import { boolean } from 'boolean';
 import nuxtConfig from '~/nuxt.config';
 
-const stringToBoolean = (value: string): boolean => value === 'True';
-const booleanToString = (value: boolean): string => (value ? 'True' : 'False');
-
-export interface DisplayPreferencesState extends DisplayPreferencesDto {
-  CustomPrefs: {
-    [key: string]: string;
-  };
+/**
+ * TODO: Generate this from the nuxtConfig file
+ */
+export enum SupportedLocales {
+  auto = 'auto',
+  cs = 'cs',
+  de = 'de',
+  enUs = 'enUs',
+  es = 'es',
+  fr = 'fr',
+  nb = 'nb',
+  nl = 'nl',
+  pl = 'pl',
+  ro = 'ro',
+  sk = 'sk',
+  sl = 'sl',
+  sv = 'sv',
+  ta = 'ta',
+  tr = 'tr',
+  vi = 'vi',
+  zhCn = 'zhCn'
 }
 
-const defaultCustomPrefs: {
-  [key: string]: string;
-} = {
-  darkMode: booleanToString(
-    nuxtConfig.vuetify?.theme?.default === 'dark' || false
-  ),
-  locale: 'auto'
-};
+/**
+ * Casted typings for the CustomPrefs property of DisplayPreferencesDto
+ */
+export interface CustomPreferences {
+  darkMode: boolean;
+  locale: SupportedLocales;
+}
+/**
+ * DisplayPreferencesDto but with our custom typings for CustomPrefs property of DisplayPreferencesDto
+ */
+export interface ClientPreferences
+  extends Omit<DisplayPreferencesDto, 'CustomPrefs'> {
+  CustomPrefs?: CustomPreferences;
+}
+/**
+ * Typings for Vuex's store state
+ */
+export interface DisplayPreferencesState extends CustomPreferences {
+  syncing: boolean;
+}
 
 const defaultState = (): DisplayPreferencesState => ({
-  CustomPrefs: defaultCustomPrefs
+  darkMode:
+    nuxtConfig.vuetify?.theme?.dark !== undefined
+      ? nuxtConfig.vuetify?.theme?.dark
+      : true,
```

Comment:
@ThibaultNocchi Why you restored this during your rebase?

Reply:
Don't worry, it's a manual commit I did. To me, we should set the default value from the config, wdyt?

Reply:
Seems fine, just didn't understand the reasoning behind but imo it's good so people building this client and changing default themes at build time doesn't get unexpected results.

Reply:
In the end we should have a config file if we take over the old PR :p